src/adapter/src/coord/cluster_scheduling.rs

```diff
@@ -0,0 +1,169 @@
+// Copyright Materialize, Inc. and contributors. All rights reserved.
+//
+// Use of this software is governed by the Business Source License
+// included in the LICENSE file.
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0.
+
+use crate::coord::{Coordinator, Message};
+use itertools::Itertools;
+use mz_catalog::memory::objects::{CatalogItem, ClusterVariant, ClusterVariantManaged};
+use mz_controller_types::ClusterId;
+use mz_ore::soft_panic_or_log;
+use mz_sql::catalog::CatalogCluster;
+use mz_sql_parser::ast::ClusterScheduleOptionValue;
+use tracing::warn;
+
+const POLICIES: &[&str] = &[REFRESH_POLICY_NAME];
+
+const REFRESH_POLICY_NAME: &str = "refresh";
+
+impl Coordinator {
+    #[mz_ore::instrument(level = "debug")]
+    /// Call each scheduling policy, gather their decisions about turning clusters on/off, and send
+    /// a message with the result.
+    pub(crate) async fn check_scheduling_policies(&mut self) {
+        // (So far, we have only this one policy.)
+        let refresh_policy_decisions = self.check_refresh_policy().await;
+
+        let internal_cmd_tx = self.internal_cmd_tx.clone();
+        if let Err(e) = internal_cmd_tx.send(Message::SchedulingDecisions(vec![(
+            REFRESH_POLICY_NAME,
+            refresh_policy_decisions,
+        )])) {
+            // It is not an error for this task to be running after `internal_cmd_rx` is dropped.
+            warn!("internal_cmd_rx dropped before we could send: {:?}", e);
+        }
+    }
+
+    async fn check_refresh_policy(&mut self) -> Vec<(ClusterId, bool)> {
+        let mut decisions: Vec<(ClusterId, bool)> = Vec::new();
+        for cluster in self.catalog().clusters() {
+            if let ClusterVariant::Managed(ref config) = cluster.config.variant {
+                match config.schedule {
+                    ClusterScheduleOptionValue::Manual => {
+                        // nothing to do, user manages it manually
+                    }
+                    ClusterScheduleOptionValue::Refresh => {
+                        let refresh_mvs = cluster
```

Comment:
I forget, do we enforce that a cluster with a "refresh every" mat view can only contain that single mat view? If so, it seems like this could be a good spot to emit a warning or enforce that in some way

Reply:
No, in the meantime we are not enforcing that, because the first user of the feature will want to have several REFRESH MVs on the same cluster, and also supporting indexes.